.jenkins/infrastructure/docker/build_linux_docker_images.Jenkinsfile

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+OECI_LIB_VERSION = env.OECI_LIB_VERSION ?: "master"
+oe = library("OpenEnclaveCommon@${OECI_LIB_VERSION}").jenkins.common.Openenclave.new()
+
+AGENTS_LABEL = params.AGENTS_LABEL
+TIMEOUT_MINUTES = params.TIMEOUT_MINUTES ?: 240
+
+INTERNAL_REPO = params.INTERNAL_REPO ?: "https://oejenkinscidockerregistry.azurecr.io"
+INTERNAL_REPO_CREDS = params.INTERNAL_REPO_CREDS ?: "oejenkinscidockerregistry"
+DOCKERHUB_REPO_CREDS = params.DOCKERHUB_REPO_CREDS ?: "oeciteamdockerhub"
+LINUX_DOCKERFILE = ".jenkins/infrastructure/docker/dockerfiles/linux/Dockerfile"
+
+def buildLinuxDockerContainers() {
+    node(AGENTS_LABEL) {
+        timeout(TIMEOUT_MINUTES) {
+            stage("Checkout") {
+                cleanWs()
+                checkout scm
+            }
+            String buildArgs = oe.dockerBuildArgs("UID=\$(id -u)", "UNAME=\$(id -un)",
+                                                  "GID=\$(id -g)", "GNAME=\$(id -gn)")
+            parallel "Build Ubuntu 18.04 Docker Image": {
+                stage("Build Ubuntu 18.04 Docker Image") {
```

Comment:
Can we further parallelize here so that it builds all containers at the same time on different hosts?